source/manual/howto-setup-maintenance-mode.html.md

```diff
@@ -0,0 +1,78 @@
+---
+owner_slack: "#govuk-developers"
+title: Setup maintenance mode
+section: Frontend
+layout: manual_layout
+parent: "/manual.html"
+related_repos: [frontend, government-frontend]
+---
+
+## What is maintenance mode?
+
+Maintenance mode is when a public-facing application displays a user-friendly message to inform users about planned downtime such as a database upgrade instead of displaying unintended error pages.
+
+## Setup maintenance mode
+
+There are two main steps involved:
+
+1. Create a maintenance flag in the app views
+2. Enable/Disable the maintenance mode
+
+## Create a maintenance flag
+
+Note: This is a one-time setup that is done in the application
+
+1. Set up the initializer file to read the environment variables
+
+Rails.application.config.maintenance_mode = ENV["MAINTENANCE_MESSAGE"].present?
+Rails.application.config.maintenance_message = ENV["MAINTENANCE_MESSAGE"]
+
+Example PR [https://github.com/alphagov/frontend/pull/5072]
+
+### In app views
+
+1. Identify the form/forms that would be impacted
+2. Set the maintenance flag for the whole form in the particular views by placing the below notice component in the else part i.e when maintenance flag is set
+
+```erb
+<% unless Rails.application.config.maintenance_mode %>
+[The actual form]
+else
+<%= render "govuk_publishing_components/components/notice", {
+    description_govspeak: sanitize(Rails.application.config.maintenance_message),
+  } %>
+<% end %>
+```
+
+### In routes file
+
+When we need to ensure that all routes serve the maintenance page during downtime, we can enable the maintenance flag in `routes.rb`. This diverts all incoming traffic to a dedicated maintenance controller, regardless of the request method (GET, POST, etc.)
+
+1. Add the below line in routes.rb file to redirect all incoming requests to the `MaintenanceController`:
+
+```erb
+match "*" => "maintenance#show", via: %i[get post] if Rails.application.config.maintenance_mode
+```
+
+2. Add the notice component as in app views in a `maintenance view` page
+
+Example PR [https://github.com/alphagov/email-alert-frontend/pull/2159]
+
+## Enable/Disable the maintenance mode
```

Comment:
Suggestion:
Update 
`Enable/Disable the maintenance mode` 
to 
`Enable/Disable the maintenance mode in govuk-helm-charts`